src/ui/ons/ONSTextField.svelte

```diff
@@ -1,15 +1,56 @@
 <script>
-  export let id;
-  export let value = "";
+  export let id, type, textFieldValue, onChange;
   export let hint = "";
+  export let inputLabel = "";
+  export let showPlaceholder = false;
+  export let errorMessage = "Custom error message";
+  export let renderErrorMsg = false;
+
+  function typeAction(node) {
+    node.type = type;
+  }
 </script>
 
-<div class="ons-field">
-  <label class="ons-label" for={id}><slot /></label>
-  {#if hint}
-    <span id="description-hint" class="ons-label__description  ons-input--with-description">
-      {hint}
-    </span>
-  {/if}
-  <input type="text" {id} class="ons-input ons-input--text ons-input-type__input" bind:value />
-</div>
+{#if renderErrorMsg}
+  <div class="ons-panel ons-panel--error ons-panel--no-title ons-u-mb-s" id="{id}-error">
+    <span class="ons-u-vh">Error: </span>
+    <div class="ons-panel__body">
+      <p class="ons-panel__error">
+        <strong>{errorMessage}</strong>
+      </p>
```

Comment:
Your proposal to have a separate <ONSError errorMessage=""></ONSError> style component was a good one. Personally I don't like the idea that if there is an error the top level div is an `ons-panel` and if there is no error it is an `ons-field`. It sounds to me as though you would be lining up a world of page alignment problems doing that but I'd like to hear from @designorant what the standard way of implementing in React would be. Either way building the ONSError component would tidy up the code here and across the design system - you could cut out the two code blocks and the code for here would become 
```
{#if renderErrorMsg}<ONSError message={errorMessage}></if>

[All the standard code]

{#if renderErrorMsg}</ONSError>{/if}
```

Reply:
This does indeed seem rather odd way of doing it but let's stick to what DS have in there. Any issues we may have with this approach can be fed back to DS.

Reply:
I have created a separate ons partial <ONSError /> component which renders conditionally based on the bool value of the renderError props and is imported in the <ONSTextField /> component.